File: tic-tac-toe-core/src/main/java/by/xxx/pupil/model/PlayerUtils.java
package by.xxx.pupil.model;

public class PlayerUtils {

    public static Player getRival(Player player) {
        switch (player) {
            case CROSSES:
                return Player.NOUGHTS;
            case NOUGHTS:
                return Player.CROSSES;
            default:
                throw new RuntimeException("Unknown player!");
        }
    }

    public static CellType getCorrespondingCellType(Player player) {
        switch (player) {
            case CROSSES:
                return CellType.CROSS;
            case NOUGHTS:
                return CellType.NOUGHT;
            default:
                throw new RuntimeException("Unknown player!");
        }
    }

}

File: tic-tac-toe-core/src/main/kotlin/by/xxx/pupil/model/PlayerUtils.kt
package by.xxx.pupil.model

fun getRival(player: Player): Player =
        when (player) {
            Player.CROSSES -> Player.NOUGHTS
            Player.NOUGHTS -> Player.CROSSES
        }

fun getCorrespondingCellType(player: Player): CellType =
        when (player) {
            Player.CROSSES -> CellType.CROSS
            Player.NOUGHTS -> CellType.NOUGHT
        }

fun resolveCellTypeFromSymbol(symbol: Char): CellType =
        when (symbol) {
            'x' -> CellType.CROSS
            '0' -> CellType.NOUGHT
            ' ' -> CellType.EMPTY
            '#' -> CellType.OBSTACLE
            else -> throw RuntimeException("Unknown symbol $symbol for cell!")
        }